File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int maxH = 0;

        while(left < right){
            int crntH = Math.min(height[left],height[right]) * (right - left);

            if(height[left] > height[right]){
                right--;
            }
            else{
                left++;
            }

            if(crntH > maxH){
                maxH = crntH;
            }
        }
        return maxH;
        
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    
    let totalCap = 0;
    let i = 0;
    let j = height.length-1;

    while(i < j){
        let maxCap = 0;
       if(height[i] > height[j]){
         maxCap = height[j] * Math.abs(i - j);
        //  console.log("i",maxCap);
        j--;
       }
       else{
        maxCap = height[i]  * Math.abs(i - j);
        // console.log("j",maxCap);
        i++;
       }

       totalCap = Math.max(totalCap, maxCap);
    //    console.log(totalCap)

    }
    return totalCap;
};
